Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -150,6 +152,14 @@ CGPoint GetStartingPoint() const {
         return startingPoint;
     }
 
+    bool GetIsFirstCall() {
+        return isFirstCall;
+    }
+
+    void SetIsFirstCall(bool firstCall) {
+        isFirstCall = firstCall;
+    }
```

Comment:
According to the docs - it is not the "first call", rather the existence of an open sub path - that should be the mechanism used here.